commands/scanandfixrepos.go

```diff
@@ -22,43 +18,22 @@ type ScanAndFixRepositories struct {
 func (saf *ScanAndFixRepositories) Run(configAggregator utils.FrogbotConfigAggregator, client vcsclient.VcsClient) error {
 	var errList strings.Builder
 	for repoNum := range configAggregator {
-		if err := saf.scanAndFixSingleRepository(&configAggregator[repoNum], client); err != nil {
-			errList.WriteString(fmt.Sprintf("repository %s returned the following error: \n%s\n", configAggregator[repoNum].RepoName, err.Error()))
-		}
+		saf.scanAndFixSingleRepository(&configAggregator[repoNum], client, errList)
 	}
 	if errList.String() != "" {
 		return errors.New(errList.String())
 	}
 	return nil
 }
 
-func (saf *ScanAndFixRepositories) scanAndFixSingleRepository(repoConfig *utils.FrogbotRepoConfig, client vcsclient.VcsClient) error {
+func (saf *ScanAndFixRepositories) scanAndFixSingleRepository(repoConfig *utils.FrogbotRepoConfig, client vcsclient.VcsClient, errList strings.Builder) {
 	for _, branch := range repoConfig.Branches {
-		shouldScan, checkedCommit, err := saf.shouldScanLatestCommit(context.Background(), repoConfig, client, branch)
-		if err != nil {
-			return err
-		}
-		if !shouldScan {
-			log.Info(fmt.Sprintf("Commit '%s' in repo '%s', branch '%s' has already been scanned. Skipping the scan.", checkedCommit, repoConfig.RepoName, branch))
-			continue
-		}
-		if err = saf.downloadAndRunScanAndFix(repoConfig, branch, client); err != nil {
+		if err := saf.downloadAndRunScanAndFix(repoConfig, branch, client); err != nil {
 			if _, isCustomError := err.(*utils.ErrUnsupportedFix); isCustomError {
-				log.Debug(err.Error())
-			} else {
-				// Scan failed, mark commit status failed with error info
-				if e := saf.setCommitBuildStatus(client, repoConfig, vcsclient.Fail, checkedCommit, fmt.Sprintf("Frogbot error: %s", err)); e != nil {
-					return errors.Join(e, err)
-				}
-				return err
+				errList.WriteString(err.Error())
```

Comment:
Let's uses `errors.Join` and have this function return an error. It's more stright forward this way.